jax/interpreters/xla.py

```diff
@@ -254,25 +259,37 @@ def write(v, node):
     _map(write, all_freevars, map(c.ParameterWithShape, freevar_shapes))
   _map(write, jaxpr.invars, map(c.ParameterWithShape, arg_shapes))
   for eqn in jaxpr.eqns:
+    # For nested jits, the outer jit sets the backend on all inner jits unless
+    # an inner-jit also has a conflicting explicit backend specification.
+    inner_backend = eqn.params.pop('backend', None)
```

Comment:
pop get

Reply:
nvm this is necessary to not have kwarg conflict...

Reply:
I don't think we should do an in-place mutating edit to a jaxpr equation. Instead of mutating the old one, you can just form a new params dict; look a few lines down for an example: `new_params = {k: eqn.params[k] for k in eqn.params if k != 'axis_name'}`.

Reply:
(You might also want to move this check under the branches where it matters, namely `elif eqn.primitive in reduction_translations` and `elif eqn.primitive in call_translations` I think.)

Reply:
How about making a helper function, like `new_params = check_backend_param(params)`? You can call that in each branch where this matters, and then that function can raise an error if needed while also producing the new params.

Reply:
done.